src/layouts/ScaleStrategies.js

```diff
@@ -21,11 +21,12 @@ define(function() {
      * Utility to offset a scale to accommodate content margins.
      *
      * @param {number} dimension - The value of the dimension to offset.
-     * @param {number} margin - The margin around the dimensions.
+     * @param {number} marginA - The margin on one side of the item.
+     * @param {number} marginB - The margin on the opposite side of the item.
      * @return {number}
      */
-    function marginScaleOffset(dimension, margin) {
-        return (dimension - margin * 2) / dimension;
+    function marginScaleOffset(dimension, marginA, marginB) {
+        return (dimension - (marginA + marginB)) / dimension;
```

Comment:
For a utility function it might be good to just return 0 instead of throwing when dimension is 0 if we have a 0 height page for some reason.